lib/passport-configurator.js

```diff
@@ -262,420 +268,585 @@ PassportConfigurator.prototype.configureProvider = function(name, options) {
+          if (link) {
+            if (req.user) {
+              self.userCredentialModel.link(
+                req.user.id,
+                provider,
+                authScheme,
+                profile,
+                {
+                  accessToken: accessToken,
+                  refreshToken: refreshToken,
+                },
+                options,
+                done
+              );
+            } else {
+              done(g.f('No user is logged in'));
+            }
           } else {
-            done(g.f('No user is logged in'));
+            self.userIdentityModel.login(
+              provider,
+              authScheme,
+              profile,
+              {accessToken: accessToken, refreshToken: refreshToken},
+              options,
+              loginCallback(req, done)
+            );
           }
-        } else {
-          self.userIdentityModel.login(provider, authScheme, profile,
-            {accessToken: accessToken, refreshToken: refreshToken},
-            options, loginCallback(req, done));
         }
-      }
       );
       break;
     case 'saml':
-      strategy = new AuthStrategy(_.defaults({
-        callbackUrl: callbackURL,
-        cert: options.certPath ? fs.readFileSync(options.certPath, 'utf-8') : options.cert,
-        privateCert: options.privateCertPath ?
-          fs.readFileSync(options.privateCertPath, 'utf-8') : options.privateCert,
-        decryptionPvk: options.decryptionPvkPath ?
-          fs.readFileSync(options.decryptionPvkPath, 'utf-8') : options.decryptionPvk,
-        passReqToCallback: true,
-      }, options),
-      function(req, profile, done) {
-        // Azure saml profile returns http://schemas.microsoft.com/identity/claims/objectidentifier
-        profile.id = profile.id || profile['http://schemas.microsoft.com/identity/claims/objectidentifier'];
-        if (link) {
-          if (req.user) {
-            self.userCredentialModel.link(req.user.id, name, authScheme,
-              profile, {}, options, done);
+      strategy = new AuthStrategy(
+        _.defaults(
+          {
+            callbackUrl: callbackURL,
+            cert: options.certPath ?
+              fs.readFileSync(options.certPath, 'utf-8') :
+              options.cert,
+            privateCert: options.privateCertPath ?
+              fs.readFileSync(options.privateCertPath, 'utf-8') :
+              options.privateCert,
+            decryptionPvk: options.decryptionPvkPath ?
+              fs.readFileSync(options.decryptionPvkPath, 'utf-8') :
+              options.decryptionPvk,
+            passReqToCallback: true,
+          },
+          options
+        ),
+        function(req, profile, done) {
+          // passport-saml returns nameID as specified in the SAML specification
+          // Azure saml profile returns http://schemas.microsoft.com/identity/claims/objectidentifier
+          profile.id =
+            profile.id ||
+            profile.nameID ||
```

Comment:
This is the actual change introduced in the PR. Rest is just related to `eslint` formatting after I rebased.